### PR TITLE
Allow mid-note automation of BitInvader sample length

### PR DESCRIPTION
**Pull request by DouglasDGI**
This is one of the things requested in #4238.  It simply updates the sample length value while in the middle of playing a note in BitInvader rather than only reading the value at the beginning of the note.  You can get some very cool sounds out of this!

### CODE REVIEW

File: plugins/bit_invader/bit_invader.cpp [hunk 1]
```diff
@@ -65,17 +65,16 @@ Plugin::Descriptor PLUGIN_EXPORT bitinvader_plugin_descriptor =
...
 	sample_realindex( 0 ),
 	nph( _nph ),
-	sample_length( _length ),
 	sample_rate( _sample_rate ),
 	interpolation( _interpolation)
 {
-	sample_shape = new float[sample_length];
-	for (int i=0; i < _length; ++i)
+	sample_shape = new float[200];
+	for (int i=0; i < 200; ++i)
```
Comment (DomClark) [inline plugins/bit_invader/bit_invader.cpp]: The `_shape` parameter contains the values from the `bitInvader`'s `m_graphModel`, but this only has an initial size of 128. The new code here could read memory out-of-bounds until the graph's length is increased to 200 for the first time. You'll want to ensure that the graph has the full 200 samples when the `bitInvader` is constructed.

**Comment by zonkmachine** [comment]
@DouglasDGI This PR will most likely be affected by how https://github.com/LMMS/lmms/issues/3797 is fixed. You may want to chip in there.

File: plugins/bit_invader/bit_invader.cpp (hunk 1 above)
Comment (zonkmachine) [reply plugins/bit_invader/bit_invader.cpp]: My OCD tells me that we should increase it to 256. Cause that's what the length would be on an old wavetable synt.h. Or a new one (http://www.doepfer.de/a112.htm).

Comment (DouglasDGI) [reply plugins/bit_invader/bit_invader.cpp]: I agree.  The maximum being at 200 and not 256 has always bugged me.

**Comment by zonkmachine** [comment]
I managed to somehow push the bitinvader into bad data while testing this PR. By judging from the graph it seem to reach infinites on the positive side. I don't know how this happened or if it's from this PR but I'm filing it here for the time being. The file loads on the stable branch too and will give the same noise.

[bitinvaderloud.mmp.zip](https://github.com/LMMS/lmms/files/2824586/bitinvaderloud.mmp.zip)

**Comment by PhysSong** [comment]
> I managed to somehow push the bitinvader into bad data while testing this PR.

I dumped the saved data. For some reason the decoded data contain Base64 data. I think the data were corrupted while converting to Base64 but not sure why.
```
00000000  d7 57 6b 3f 2a cf 14 bf  98 c9 4a be 90 79 47 3e  |.Wk?*.....J..yG>|
00000010  16 bb 0a 3f 45 67 22 3f  d8 19 6a be 58 25 ac 3e  |...?Eg"?..j.X%.>|
00000020  b0 59 57 be dc 9f 6d bf  f4 b9 79 3e 43 97 21 3f  |.YW...m...y>C.!?|
00000030  9a b5 cc 3e c8 43 64 3f  6c 43 36 3f f5 a7 7a 3f  |...>.Cd?lC6?..z?|
00000040  8a 0b 45 bf d6 83 6a bf  00 ed 80 be d2 47 69 bf  |..E...j......Gi.|
00000050  0c 2d 86 be d8 f7 6b 3f  30 45 98 3e 6a 13 35 3f  |.-....k?0E.>j.5?|
00000060  94 b7 49 3f 36 eb 1a 3f  21 3d 01 00 00 00 00 00  |..I?6..?!=......|
00000070  78 1b fe ab 6e 7f 00 00  78 1b fe ab 6e 7f 00 00  |x...n...x...n...|
00000080  00 00 00 00 8c 00 00 00  e8 00 00 00 00 00 00 00  |................|
00000090  18 00 00 00 00 00 00 00  b1 40 01 00 00 00 00 00  |.........@......|
000000a0  78 7b 35 ac 36 7f 00 00  78 7b 35 ac 36 7f 00 00  |x{5.6...x{5.6...|
000000b0  43 6a 39 46 5a 79 49 2f  32 42 6c 71 76 6c 67 6c  |Cj9FZyI/2Blqvlgl|
000000c0  72 44 36 77 57 56 65 2b  33 4a 39 74 76 2f 53 35  |rD6wWVe+3J9tv/S5|
000000d0  65 54 35 44 6c 79 45 2f  6d 72 58 4d 50 73 68 44  |eT5DlyE/mrXMPshD|
000000e0  5a 44 39 73 51 7a 59 2f  39 61 64 36 50 34 6f 4c  |ZD9sQzY/9ad6P4oL|
000000f0  52 62 2f 57 67 32 71 2f  41 4f 32 41 76 74 4a 48  |Rb/Wg2q/AO2AvtJH|
00000100  61 62 38 4d 4c 59 61 2b  32 50 64 72 50 7a 42 46  |ab8MLYa+2PdrPzBF|
00000110  6d 44 35 71 45 7a 55 2f  d1 02 00 00 00 00 00 00  |mD5qEzU/........|
00000120  78 5b da 91 09 7f 00 00  70 26 92 03 00 00 00 00  |x[......p&......|
00000130  e0 c3 69 92 09 7f 00 00  00 00 00 00 00 00 00 00  |..i.............|
*
00000190  e0 c3 69 92 09 7f 00 00  09 fa 74 bf 08 09 71 bf  |..i.......t...q.|
000001a0  5d 83 6c bf d5 6b 67 bf  98 c5 61 bf 19 94 5b bf  |].l..kg...a...[.|
000001b0  2f db 54 bf 03 9f 4d bf  03 e4 45 bf f7 ae 3d bf  |/.T...M...E...=.|
000001c0  ef 04 35 bf 4a eb 2b bf  97 67 22 bf bc 7f 18 bf  |..5.J.+..g".....|
000001d0  db 39 0e bf 3c 9c 03 bf  e3 5a f1 be 76 e8 da be  |.9..<....Z..v...|
000001e0  15 ef c3 be cf 7c ac be  28 a0 94 be cf cf 78 be  |.....|..(.....x.|
000001f0  bc c5 47 be 74 40 16 be  04 bd c8 bd 26 fb 48 bd  |..G.t@......&.H.|
00000200  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
*
000002c4
```

**Comment by PhysSong** [comment]
Anyway, is this ready for merge?

**Comment by DouglasDGI** [comment]
As long as the corruption problem mentioned above isn't related to these changes, it is ready for merge.

**Pull request merged**